__tests__/cache/cache-manager.test.ts

```diff
@@ -0,0 +1,181 @@
+import { CacheManager, InMemoryCache } from '../../src/cache';
+import { CacheEntry, CacheKey } from '../../src/cache/shared';
+import {
+  TEST_ACCESS_TOKEN,
+  TEST_AUDIENCE,
+  TEST_CLIENT_ID,
+  TEST_ID_TOKEN,
+  TEST_SCOPES,
+  dayInSeconds,
+  nowSeconds
+} from '../constants';
+
+describe('CacheManager', () => {
+  let manager: CacheManager;
+
+  beforeEach(() => {
+    manager = new CacheManager(new InMemoryCache().enclosedCache);
+  });
+
+  it('returns undefined from the cache when expires_in < expiryAdjustmentSeconds', async () => {
+    const data = {
+      client_id: TEST_CLIENT_ID,
+      audience: 'the_audience',
+      scope: TEST_SCOPES,
+      id_token: TEST_ID_TOKEN,
+      access_token: TEST_ACCESS_TOKEN,
+      expires_in: 40,
+      decodedToken: {
+        claims: {
+          __raw: TEST_ID_TOKEN,
+          exp: nowSeconds() + dayInSeconds,
+          name: 'Test'
+        },
+        user: { name: 'Test' }
+      }
+    };
+
+    await manager.set(data);
+
+    expect(
+      await manager.get(
+        new CacheKey({
+          client_id: TEST_CLIENT_ID,
+          audience: 'the_audience',
+          scope: TEST_SCOPES
+        }),
+        60
+      )
+    ).toBeFalsy();
+  });
+
+  describe('when refresh tokens are used', () => {
+    it('strips everything except the refresh token when expiry has been reached', async () => {
+      const now = Date.now();
+      const realDateNow = Date.now.bind(global.Date);
+
+      const data = {
+        client_id: TEST_CLIENT_ID,
+        audience: 'the_audience',
+        scope: TEST_SCOPES,
+        id_token: TEST_ID_TOKEN,
+        access_token: TEST_ACCESS_TOKEN,
+        refresh_token: 'refreshtoken',
+        expires_in: dayInSeconds,
+        decodedToken: {
+          claims: {
+            __raw: TEST_ID_TOKEN,
+            name: 'Test',
+            exp: nowSeconds() + dayInSeconds * 2
+          },
+          user: { name: 'Test' }
+        }
+      };
+
+      await manager.set(data);
+
+      const cacheKey = new CacheKey({
+        client_id: TEST_CLIENT_ID,
+        audience: 'the_audience',
```

Comment:
I wonder if this should be extracted to a constant as well? you are using the raw string on every test on this diff